src/harness/testcases/WINNF_FT_S_IPR_testcase.py

```diff
@@ -0,0 +1,220 @@
+      test_harnesses.append(test_harness)
+
+    # Register N2 CBSDs with the SAS UUT and request Grants.
+    n2_domain_proxy = DomainProxy(self)
+    n2_domain_proxy.registerCbsdsAndRequestGrants(
+        config['registrationRequestsN2'],
+        config['grantRequestsN2'],
+        conditional_registration_data=config['conditionalRegistrationDataN2'])
+
+    # Trigger CPAS.
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Register N3 CBSDs with the SAS UUT and request Grants.
+    n3_domain_proxy = DomainProxy(self)
+    n3_domain_proxy.registerCbsdsAndRequestGrants(
+        config['registrationRequestsN3'],
+        config['grantRequestsN3'],
+        conditional_registration_data=config['conditionalRegistrationDataN3'])
+
+    # Heartbeat all SAS UUT CBSDS
+    n2_domain_proxy.heartbeatForAllActiveGrants()
+    n3_domain_proxy.heartbeatForAllActiveGrants()
+
+    # None of the N3 CBSDS should be authorized.
+    for cbsd in n3_domain_proxy.getCbsdsWithAtLeastOneAuthorizedGrant():
+      for operationParam in cbsd.getOperationParamsOfAllAuthorizedGrants():
+        # By definition the N3 CBSDs are in a DPA neighborhood. Check frequency.
+        self.assertFalse(
+            self.frequencyInBand(
+                operationParam['operationFrequencyRange']['lowFrequency'],
+                operationParam['operationFrequencyRange']['highFrequency']),
+            msg='CBSD (cbsd_id=%s, fcc_id=%s, sn=%s) '
+            'is authorized after IPR.1 step 7. SAS UUT FAILS this test. '
+            '(If this config is new please verify the CBSD is in a DPA)' %
+            (cbsd.getCbsdId(), cbsd.getRegistrationRequest()['fccId'],
+             cbsd.getRegistrationRequest()['cbsdSerialNumber']))
+
+    # Get SAS UUT FAD and Test Harness FADs.
+    sas_uut_fad = getFullActivityDumpSasUut(self._sas, self._sas_admin)
+    test_harness_fads = []
+    for test_harness in test_harnesses:
+      test_harness_fads.append(
+          getFullActivityDumpSasTestHarness(
+              test_harness.getSasTestHarnessInterface()))
+
+    # Trigger CPAS.
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Initialize DPA objects and calculate movelist for each DPA.
+    # TODO(taran): Update once DPA movelist reference model changes are reviewed.
+    dpas = [Dpa(dpa) for dpa in config['dpas']]
+    for dpa in dpas:
+      dpa.SetGrantsFromFad(sas_uut_fad, test_harness_fads)
+      dpa.CalcMoveLists()
+    # Heartbeat SAS UUT grants.
+    n2_domain_proxy.heartbeatForAllActiveGrants()
+    n3_domain_proxy.heartbeatForAllActiveGrants()
+    # Get CbsdGrantInfo list of SAS UUT grants that are in an authorized state.
+    grant_info = getGrantsFromDomainProxies([n2_domain_proxy, n3_domain_proxy])
+
+    # Check grants do not exceed each DPAs interference threshold.
+    for dpa in dpas:
+      self.assertTrue(dpa.CheckInterference(channel=(3550, 3650), sas_uut_active_grants=grant_info, margin_db=10))
```

Comment:
You have to give a channel which is 10MHz wide. The test will test that specific channel aggregated interference.

Reply:
Now pulls from the value from the config. Would it be possible to get a check interference function that uses the frequency ranges set in SetFreqRange?

Reply:
I can do that. Let me sync with you